dot/state/block.go

```diff
@@ -734,3 +745,8 @@ func (bs *BlockState) StoreRuntime(hash common.Hash, rt runtime.Instance) {
 func (bs *BlockState) GetNonFinalisedBlocks() []common.Hash {
 	return bs.bt.GetAllBlocks()
 }
+
+// GenerateTrieProof returns the proofs related to the keys on the state root trie
+func (bs *BlockState) GenerateTrieProof(stateRoot common.Hash, keys [][]byte) ([][]byte, error) {
+	return trie.GenerateProof(stateRoot[:], keys, bs.db)
+}
```

Comment:
this would make more sense in the `storageState`

Reply:
Updated